api/src/test/java/org/openmrs/module/attachments/obs/TestHelper.java

```diff
@@ -198,4 +239,167 @@ public String getComplexObsDir() {
 	public String getTestComplexObsFilePath() {
 		return getComplexObsDir() + "/" + getTestFileNameWithExt();
 	}
+	
+	/**
+	 * Boilerplate method to save a collection of complex obs.
+	 *
+	 * @param count The number of the complex obs to be saved.
+	 */
+	public List<Obs> saveComplexObs(int count) throws IOException {
+		
+		init();
+		
+		List<Obs> obsList = new ArrayList<>();
+		byte[] randomData = new byte[20];
+		
+		Patient patient = Context.getPatientService().getPatient(2);
+		Visit visit = Context.getVisitService().getVisit(1);
+		
+		EncounterService encounterService = Context.getEncounterService();
+		EncounterType encounterType = encounterService.getEncounterType(1);
+		Provider provider = Context.getProviderService().getProvider(1);
+		context.getAdministrationService().saveGlobalProperty(
+		    new GlobalProperty(AttachmentsConstants.GP_ENCOUNTER_TYPE_UUID, encounterType.getUuid()));
+		Encounter encounter = context.getAttachmentEncounter(patient, visit, provider);
+		
+		for (int i = 0; i < count; i++) {
+			String fileCaption = RandomStringUtils.randomAlphabetic(12);
+			new Random().nextBytes(randomData);
+			MockMultipartFile multipartRandomFile = new MockMultipartFile(String.valueOf(i), String.valueOf(i),
+			        "application/octet-stream", randomData);
+			obsList.add(obsSaver.saveOtherAttachment(visit, patient, encounter, fileCaption, multipartRandomFile,
+			    ValueComplex.INSTRUCTIONS_DEFAULT));
+		}
+		return obsList;
+	}
+	
+	/**
+	 * Boilerplate method to save a collection of complex obs.
+	 *
+	 * @param count The number of the complex obs to be saved.
+	 */
+	public List<Obs> saveComplexObs(Encounter encounter, int count, int otherCount, boolean allComplexRelatedToAttach)
+	        throws IOException {
+		init();
+		
+		// Creating the list of Complex Obs relevnt to the attachment module (
+		// Attachments ) using ComplexObsSaver class.
+		List<Obs> obsList = new ArrayList<>();
+		List<Obs> otherObsList = new ArrayList<>();
+		byte[] randomData = new byte[20];
+		Patient patient = encounter.getPatient();
+		Visit visit = encounter.getVisit();
+		for (int i = 0; i < count; i++) {
+			String fileCaption = RandomStringUtils.randomAlphabetic(12);
+			new Random().nextBytes(randomData);
+			MockMultipartFile multipartRandomFile = new MockMultipartFile(String.valueOf(i), String.valueOf(i),
+			        "application/octet-stream", randomData);
+			obsList.add(obsSaver.saveOtherAttachment(visit, patient, encounter, fileCaption, multipartRandomFile,
+			    ValueComplex.INSTRUCTIONS_DEFAULT));
+		}
+		for (int i = 0; i < otherCount; i++) {
+			// Creating the complex Obs not relevent / relevent to the attachment module
+			// during the same encounter.
+			Obs obs = new Obs();
+			if (allComplexRelatedToAttach) {
```

Comment:
I don't understand this `if` statement, why don't you just loop `otherCount` times to create as many non-attachments related complex obs as specified using `otherConceptComplex` as an underlying concept complex?

That was simply the plan, did I miss something?

Reply:
Initially you mentioned that we should able to configure that non-attachments to attachment related and check with the AttachmentService. 

"Edit the configuration and encompass one (or two) of the other complex obs to be also Attachments-managed.
Verify that the service returns the two original Attachments-managed ones plus the one (or two) that were encompassed through the configuration change. "

That's why I used that boolean varibale for configure that it is non-attachment or attachment related. Is It ok ? ( or is it a wrong way ?)